### PR TITLE
Reduce number of requests in header-values-normalize

### PR DESCRIPTION
**Pull request by annevk**
See discussion in https://github.com/w3c/web-platform-tests/pull/4525.

### CODE REVIEW

File: fetch/api/headers/header-values-normalize.html [hunk 1]
```diff
@@ -22,43 +22,49 @@
...
-          assert_equals(res.headers.get("x-request-value-test"), useExpectedVal)
-        })
-      }
-    }, "fetch() with value " + encodeURI(useVal))
-  }
+  promise_test((t) => {
+    if(fail) {
+      promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val1": val1} }))
+      promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val2": val2} }))
+      return promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val3": val3} }))
```
Comment (annevk) [inline fetch/api/headers/header-values-normalize.html]: I'm not sure about the above three lines being the correct way to do this. I asked in #whatwg but nobody was around to answer.

Comment (mkruisselbrink) [inline fetch/api/headers/header-values-normalize.html]: You probably want that to be `return Promise.all([promise_rejects(...), promise_rejects(...), promise_rejects(...)])`?

Comment (annevk) [inline fetch/api/headers/header-values-normalize.html]: Does that work? I cannot really find any documentation about how this is supposed to work. @jgraham?

Comment (jdm) [inline fetch/api/headers/header-values-normalize.html]: Yeah, the callback to promise_test needs to return a single promise that encompasses all asynchronous behaviour. Promise.all is what you want here.

Comment (annevk) [inline fetch/api/headers/header-values-normalize.html]: Thanks, fixed.

**Comment by annevk** [comment]
@ricea does this look good now? @jdm?

**Comment by jdm** [comment]
Yep!

**Comment by ricea** [comment]
Looks good to me.

**Pull request merged**